packages/wallet-sdk/src/core/provider/interface.ts

```diff
@@ -42,10 +42,47 @@ export interface AppMetadata {
   appChainIds: number[];
 }
 
-export interface Preference {
+type PostOnboardingAction = 'none' | 'onramp' | 'magicspend';
+
+type OnrampPrefillOptions = {
+  contractAddress?: string;
+  amount: string;
+  chainId: number;
+};
+
+export type Preference = {
   options: 'all' | 'smartWalletOnly' | 'eoaOnly';
+  /**
+   * @deprecated this is only for internal use
+   */
   keysUrl?: string;
-}
+  /**
+   * @param postOnboardingAction
+   * @type {PostOnboardingAction}
+   * @description Recommends the action to be taken after the onboarding process to engage users with various flows. Possible values:
+   * - `none`: No action is recommended post-onboarding. (Default experience)
+   * - `onramp`: Recommends initiating the onramp flow, allowing users to prefill their account with a specified asset, chain, and amount.
+   * - `magicspend`: Suggests linking the users retail Coinbase account for seamless transactions.
+   */
+  postOnboardingAction?: PostOnboardingAction;
+  /**
+   * @param onrampPrefillOptions
+   * @type {OnrampPrefillOptions}
+   * @description This option only functions when `postOnboardingAction` is set to `onramp`.
+   * - Prefills the onramp flow with the specified asset, chain, and suggested amount,
+   * allowing users to prefill their account.
+   * - Ensure the asset and chain are supported by the onramp provider
+   * (e.g., Coinbase Pay - CBPay).
+   */
```

Comment:
add: defaults to an asset selector with 0 as the initial amount

Can we also add some help for devs in determining the right formats here, and determining what is supported by CBPay?

Reply:
updated with a link from cbpay docs